utils/build-script

```diff
@@ -147,6 +156,22 @@ def print_xcodebuild_versions(file=sys.stdout):
     print(fmt.format(version=version, sdks=sdks), file=file)
     file.flush()
 
+def validate_xcode_compatibility():
+    if sys.platform != 'darwin':
+        return
+
+    if os.getenv("SKIP_XCODE_VERSION_CHECK"):
+        print("note: skipping Xcode version check")
+        return
+
+    version = shell.capture(
+        ['xcodebuild', '-version'], dry_run=False, echo=False).strip()
+    if not version.endswith(_SUPPORTED_XCODE_BUILDS):
+        raise SystemExit(
+            "error: using unsupported Xcode version:\n\n{}\n\nInstall one of: {}".format(
```

Comment:
Would be good to add "or set SKIP_XCODE_VERSION_CHECK to skip this check at your own risk" so the workaround is immediately visible.

Reply:
Done!